datafusion/sqllogictest/test_files/string/init_data.slt.part

```diff
@@ -23,6 +23,8 @@ create table test_source as values
   ('under_score', 'un_____core', 'un iść core', 'chrząszcz na łące w 東京都'),
   ('percent', 'p%t', 'pan Tadeusz ma iść w kąt', 'Pan Tadeusz ma frunąć stąd w kąt'),
   ('', '%', '', ''),
+  ('%', '\%', '', ''),
+  ('_', '\_', '', ''),
```

Comment:
This feels duplicate with what i added in string_literal, but adds test coverage  for all string variants